src/st/no3/e_bone_scimitar.c

```diff
@@ -1,35 +1,276 @@
+        InitializeEntity(g_BoneScimitarInit);
+        if (self->params != 0) {
+            self->palette += self->params;
+            self->flags &= ~(FLAG_DESTROY_IF_BARELY_OUT_OF_CAMERA |
+                             FLAG_DESTROY_IF_OUT_OF_CAMERA | 0xC00);
+            self->ext.boneScimitar.initialX =
+                g_Tilemap.scrollX.i.hi + self->posX.i.hi;
+            if (self->params & D_80183B50) {
+                DestroyEntity(self);
+                return;
+            }
+        }
+        self->ext.boneScimitar.attackTimer = 80; // Skeleton attack timer cycle
+        self->ext.boneScimitar.facingLeft = 0;   // Facing init
+        self->ext.boneScimitar.attackTimerIndex =
+            0; // Skeleton attack timer selector
+        break;
+
+    case BONE_SCIMITAR_IDLE:
+        if (func_801C5074(&D_80183C20) != 0) {
+            self->step++;
+            if (self->params != 0) {
+                SetStep(BONE_SCIMITAR_SPECIAL);
+            }
+        }
+        break;
+
+    case BONE_SCIMITAR_WALK_TOWARDS_PLAYER:
+        if (AnimateEntity(D_80183B54, self) == 0) {
+            self->facingLeft = (GetSideToPlayer() & 1) ^ 1;
+        }
+        self->ext.boneScimitar.facingLeft = self->facingLeft;
+
+        if (self->ext.boneScimitar.facingLeft == 0) {
+            self->velocityX = FIX(-0.5);
+        } else {
+            self->velocityX = FIX(0.5);
+        }
+
+        if (GetDistanceToPlayerX() < 76) {
+            self->step = BONE_SCIMITAR_WALK_AWAY_FROM_PLAYER;
+        }
+        func_801D59D0();
+        break;
+
+    case BONE_SCIMITAR_WALK_AWAY_FROM_PLAYER:
+        if (AnimateEntity(D_80183B64, self) == 0) {
+            self->facingLeft = (GetSideToPlayer() & 1) ^ 1;
+        }
+        self->ext.boneScimitar.facingLeft = self->facingLeft ^ 1;
+
+        if (self->ext.boneScimitar.facingLeft == 0) {
+            self->velocityX = FIX(-0.5);
+        } else {
+            self->velocityX = FIX(0.5);
+        }
+
+        if (GetDistanceToPlayerX() > 92) {
+            self->step = BONE_SCIMITAR_WALK_TOWARDS_PLAYER;
+        }
+        func_801D59D0();
+        break;
+
+    case BONE_SCIMITAR_ATTACK:
+        animStatus = AnimateEntity(D_80183B74, self);
+        if (self->animCurFrame == 12) {
+            self->hitboxWidth = 20;
+            self->hitboxHeight = 17;
+            self->hitboxOffX = -11;
+            self->hitboxOffY = -14;
+        } else {
+            self->hitboxWidth = 8;
+            self->hitboxHeight = 18;
+            self->hitboxOffX = -1;
+            self->hitboxOffY = 0;
+        }
+
+        if (self->animFrameIdx == 7 && self->animFrameDuration == 0) {
+            PlaySfxPositional(SFX_BONE_SWORD_SWISH_B);
+        }
+
+        if (animStatus == 0) {
+            SetStep(BONE_SCIMITAR_WALK_AWAY_FROM_PLAYER);
+            self->ext.boneScimitar.attackTimer =
+                D_80183C18[self->params % 2]
+                          [(++self->ext.boneScimitar.attackTimerIndex) & 3];
```

Comment:
This looks like it was copied from the original, but it's s a bit terse for my taste. The increment of `attackTimerIndex` isn't related to the operation of looking up and assigning the attack timer, so my preference would be that they were separated.

I'd rather get the coverage, however, so if nothing else is blocking feel free to ignore.

Reply:
Good call, agreed! Updated to split the increment and access into two lines.